### DISCUSSION TITLE
Feature Request: WearOS xDrip+ Watchface Change Colors for Ranges / Main Number

### DISCUSSION THREAD

**Discussion by potatonixlimegoggles** [NightscoutFoundation/xDrip#2996] [Ideas]
I've run into the issue that I can't adjust the colors for the ranges (and main glucose number) on my xDrip+ watchface. I'd like to make my watchface match my main xDrip+ app colors, and that isn't an option, once you move the main Android display beyond the red/white/yellow standard colors of the watchface. I'd also like the main number color to be adjustable. I wouldn't need a huge number of colors--maybe just the 10 "standard" MS Office colors, plus white, or maybe ROYGBIV + white?

**Comment by Navid200** [comment]
We don't have an active developer supporting WearOS.
I am hoping that an experienced developer would move WearOS out of xDrip.  It will need a service to be added to xDrip to establish a communication protocol between xDrip and a companion app.
We already have several of them and actually, there may be an existing one that can be used.
I personally have 0 experience.  So, I cannot do this.

If that happens, all WearOS feature requests can be addressed by tweaks to the companion app rather than to xDrip.

**Reply by potatonixlimegoggles** [reply]
That's too bad. I really like the xDrip+ WearOS app. It does everything I need it to do, including the ability to document treatments from the watchface, and even communicate directly with my Dexcom G6. I fear that spinning it off as a separate app would mean that we could lose things like the treatment documentation and direct communication that make the existing app so unique and valuable. I wouldn't want to see it become just another fancy watchface that relies on underlying complications.

**Reply by Navid200** [reply]
This is just my personal opinion.  It is NOT an official xDrip team position.
We used to have MiBand supported directly by xDrip.

Since WatchDrip was created, those who use MiBand are getting everything they used to get and more.  So, it has not been a step back.  it has been a step forward.
My hope is that something similar can happen for WearOS.

There is too much to support in xDrip and I don't want the lead developer to be loaded with more and more tasks.  It is not humanly possible to do that.
What I am suggesting is meant to allow WearOS shortcomings in the current xDrip to be addressed without needing too much change to the xDrip code.  I want the xDrip WearOS support to flourish instead of die.

**Comment by gdc27health** [comment]
What companion app do you make mention of?

**Reply by Navid200** [reply]
https://github.com/bigdigital/watchdrip

**Comment by mprosenberg** [comment]
I can send you the code. It is in lua, so it works on Tizen, and should work on Wear OS. Are you able to modify your current watch face using WatchMaker? If so open edit watch face, go to the panel you want to change, e.g. glucose, and then paste the following code into the color of the panel. If you want more, or different colors just follow the lua code format and put in the color code. 

(({txglucose} > 60) and  ({txglucose} < 130))  and '00cc00' or (({txglucose} > 129)  and ({txglucose} < 155)) and 'ccff99'  or (({txglucose} > 155)  and  ({txglucose} <165))  and 'ffff00' or  (({txglucose} > 164)  and ({txglucose} < 195))   and 'ff9966' or (({txglucose} > 194) and ({txglucose} < 241))  and 'f57842' or 'ff0000' 

Here is an example of what the watch face will look like. I pasted in the code in each of the components showing red.
![Screenshot_20201214-152409_WatchMaker](https://github.com/NightscoutFoundation/xDrip/assets/58182423/cdab3357-5cc3-4650-9f18-af3a302e3c69)

Mike

**Comment by gdc27health** [comment]
Does this support all wearOS?

What is the official stance on wearOS?
